### PR TITLE
Cascader value type change to string[] | number[]

### LINKED ISSUE #17407: cascader 中 value 添加支持  number 类型
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### 问题描述：

cascader 组件中 value 只支持string，不支持 number

### 需求

支持 number 类型

### 相关代码：

https://github.com/ant-design/ant-design/blob/master/components/cascader/index.tsx#L60

### 如果是设计如此：

     请告诉一下我为什么设计成只支持 string，谢谢

### What does the proposed API look like?

支持 number

### PR DESCRIPTION
**Pull request by hengkx**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [x] Site / document update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Solution
close #17407

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Cascader value props change to any        |
| 🇨🇳 Chinese |  Cascader   value 属性改为any       |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/cascader/index.en-US.md](https://github.com/hengkx/ant-design/blob/fix-17407/components/cascader/index.en-US.md)
[View rendered components/cascader/index.zh-CN.md](https://github.com/hengkx/ant-design/blob/fix-17407/components/cascader/index.zh-CN.md)

### CODE REVIEW

File: components/cascader/index.tsx [hunk 1]
```diff
@@ -12,21 +12,21 @@ import LocaleReceiver from '../locale-provider/LocaleReceiver';
 import warning from '../_util/warning';
 
 export interface CascaderOptionType {
-  value?: string;
+  value?: any;
```
Comment (afc163) [inline components/cascader/index.tsx]: string | number 就行了，any 对于字面量对象组件会有问题。

Comment (hengkx) [reply components/cascader/index.tsx]: 我看这个地方是`any` ,https://github.com/react-component/cascader/blob/master/index.d.ts#L4
需要一起调整吗？

**Comment by hengkx** [comment]
@afc163 你好 [https://github.com/ant-design/ant-design/blob/master/components/cascader/index.tsx#L320](https://github.com/ant-design/ant-design/blob/master/components/cascader/index.tsx#L320) 级联选择应该不能多选把，当初这样写 是为什么？  
这样相当于 value的声明应该变成 `string[] | number[] | sting[][] | number[][]`
image

File: components/cascader/index.tsx (hunk 1 above)
Comment (afc163) [reply components/cascader/index.tsx]: 一起调整下吧

Comment (ztplz) [reply components/cascader/index.tsx]: 先加吧  原来太仓促了 以后我有时间用ts重写下组件

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/pull/2776/files#diff-3141ae7cf791cd0b5edcad2c51dd6dddR135

不了解初衷。@benjycui

**Comment by simdd** [comment]
这个最后解决了吗？@hengkx

**Comment by hengkx** [comment]
没人回复 我也没关注了 @simdd

**Pull request closed without merging**